internal/querycoordv2/balance/rowcount_based_balancer.go

```diff
@@ -137,121 +141,82 @@ func (b *RowCountBasedBalancer) balanceReplica(replica *meta.Replica) ([]Segment
 			)
 			stoppingNodesSegments[nid] = segments
 		} else {
-			nodesSegments[nid] = segments
+			onlineNodesSegments[nid] = segments
 		}
 
-		cnt := 0
 		for _, s := range segments {
-			cnt += int(s.GetNumOfRows())
+			totalCnt += int(s.GetNumOfRows())
 		}
-		nodesRowCnt[nid] = cnt
-		totalCnt += cnt
 	}
 
 	if len(nodes) == len(stoppingNodesSegments) {
 		// no available nodes to balance
 		return nil, nil
 	}
 
-	if len(nodesSegments) == 0 {
-		return nil, nil
-	}
-
-	average := totalCnt / len(nodesSegments)
-	neededRowCnt := 0
-	for nodeID := range nodesSegments {
-		rowcnt := nodesRowCnt[nodeID]
-		if rowcnt < average {
-			neededRowCnt += average - rowcnt
-		}
-	}
-
-	if neededRowCnt == 0 {
+	if len(onlineNodesSegments) == 0 {
 		return nil, nil
 	}
 
 	segmentsToMove := make([]*meta.Segment, 0)
+	for _, stopSegments := range stoppingNodesSegments {
+		segmentsToMove = append(segmentsToMove, stopSegments...)
+	}
 
-	stopSegments, cnt := b.collectionStoppingSegments(stoppingNodesSegments)
-	segmentsToMove = append(segmentsToMove, stopSegments...)
-	neededRowCnt -= cnt
-
-	// select segments to be moved
-outer:
-	for nodeID, segments := range nodesSegments {
-		rowcnt := nodesRowCnt[nodeID]
-		if rowcnt <= average {
-			continue
-		}
+	// find nodes with less row count than average
+	nodesWithLessRow := newPriorityQueue()
+	average := totalCnt / len(onlineNodesSegments)
+	for node, segments := range onlineNodesSegments {
 		sort.Slice(segments, func(i, j int) bool {
 			return segments[i].GetNumOfRows() > segments[j].GetNumOfRows()
 		})
 
+		rowCount := 0
 		for _, s := range segments {
-			if rowcnt-int(s.GetNumOfRows()) < average {
+			rowCount += int(s.GetNumOfRows())
+			if rowCount <= average {
```

Comment:
This break the policy of the algorithm, in the original verison, the algorithm always balance the larger segment first, but this new policy not work like that

Reply:
You should reverse the order of the sorted slice